src/utils.jl

```diff
@@ -242,13 +247,21 @@ function read_input(file_name::AbstractString)
         end
     end
 
-    change_profile!(market_data, opt_data)
+    change_profile!(gen_data,opt_data)
+
+    for c_name in keys(market_data)
+        change_profile!(market_data[c_name], opt_data)
+    end
 
     for u_name in keys(users_data)
         comp_dict = components(users_data[u_name])
         if !isnothing(comp_dict)
             for c_name in keys(comp_dict)
-                change_profile!(comp_dict[c_name], opt_data)
+                if c_name == "market"
+                    print(u_name)
+                else
+                    change_profile!(comp_dict[c_name], opt_data)
+                end 

```

Comment:
This if may not be needed anymore: the components function shall take care of that